### ISSUE TITLE
Update Procedure & Plex Pass

### ISSUE THREAD

**Issue by vvhof** [timhaak/docker-plex#9]
Hello Tim,
maybe you can think about including an update procedure for your start.sh. This would of course mean undermining the current procedure, where dockerhub generates complete images, in the sense that they do not require excessive (119MB for Plex) reloading during runtime from apt. Debatable what would be preferable.

If you would decide on an update procedure, enabling plexpass support is trivial.

- [x] update mechanism - **won't fix**
- [ ] plexpass channel support

**Comment by vvhof** [comment]
I have given an example in #10

**Comment by timhaak** [comment]
Not quiet totally sure what you mean by an update procedure?

Do you mean running an apt update in the running container? 

Also when you say plexpass support do you mean the builds? Just checking as I have plexpass active one mine.

If you mean updating in the container a big problem is what happens if you have to restart the container. Basically you would revert back by how ever long ago that you last updated.

On my side I just pretty much do a pull restart about once a week. The restart is very quick and everything keeps working. So not sure what this would add.

The fact that it now has supervisord in the start would allow for this partially. 

If you would like to send a pull request with what you mean it might help me understand what your actually asking for.

**Comment by timhaak** [comment]
Ah wait I see you sort of did send then close then ? Busy looking

**Comment by timhaak** [comment]
Hmm ok looking at the patch you sent then closed.

First no need for "echo "deb http://shell.ninthgate.se/packages/debian plexpass main" > /etc/apt/sources.list.d/plexmediaserver.list"

This is added in the docker file.

I must say I don't like doing installs and updates in the running container as you loose one of the things that I like about docker which is repeatability. Ie If wer both running the same image it should run exactly the same. Now you would have no idea about what version your running.

I would just do a new image based on the current image that did a apt update then install of plex (would do an update basically). Would allow for a quicker build of the plex image if that's what your after.

Could you tell me what the problem is that your actually trying to solve?

**Comment by vvhof** [comment]
Hello Tim, 
Yes, i closed the pull request because i wanted to discuss it first. 

Basically i wanted to get the plexpass only versions.

1) So the ```echo "deb http://shell.ninthgate.se/packages/debian squeeze main" > /etc/apt/sources.list.d/plexmediaserver.list``` actually does not get the plexpass only release, though you can of course activate plexpass feature in non exclusive version. (c.f. http://shell.ninthgate.se/packages/debian/dists/)

2) There are two ways to go about this: 
Have one Dockerfile for the "stable" release channel.
Have one Dockerfile for the "plexpass" release channel.

OR

Integrate a flag (e.g.) "plexpass" for docker, which, if set, updates to plexpass upon start.sh.

**Comment by timhaak** [comment]
Hi
Sorry for the delayed reply got a bit busy with the rest of life.

I have some ideas about this but haven't had time to actually sit down and think about it.

Hoping to get some time this weekend once caught up with the rest of life.

Most likely will be different images. One for plex pass and two for simple upgrades that use the others as base. 

Not totally sure if I'm happy about the upgrade stuff but going to think and see what I can do.

**Comment by timhaak** [comment]
Oh yes my one concern with the plex pass is if the plex people might get upset as normally you need the plex pass to get the build.

If I see it causing a problem for them and they are even slightly unhappy I'm going to pull it.

Reason is I really like what they are doing :) and don't want to ever mess with them.

**Comment by vvhof** [comment]
hi @timhaak 
so maybe it could work like this then: https://registry.hub.docker.com/u/vvhof/plex/

**Comment by vvhof** [comment]
One question remains though: If we do this in one *docker hub* repository, we cannot have the /plexpass/Dockerfile rely on /Dockerfile, as building is done in parallel by the hub. So we might require a second docker hub repository for building the second image (from the githubrepo/plexpass/Dockerfile) and set it up to build once the first repository (from the githubrepo/Dockerfile) is built by the hub.
- [ ] create secondary docker hub repository for plexpass
- [ ] adjust README.md to reflect aforementioned change

**Comment by timhaak** [comment]
Hi

Ok threw something together quickly. Been a bit busy here and only going to quiet down in a tiny while so  not going to be able to put much time in it for now.

Ok first thing created a new repo for plexpass and added it to docker index. Should appear at `docker pull timhaak/plexpass`.

I've created to more git repo's with docker files that will use the plex or plexpass images respectively and do an update on each. I'm not going to add them to the docker index as I think they are not best practice ie you loose repeatability. But I will be first to admit you should do it the way you like to  :).

So if you want to use it will require you to do the git pull and build locally but should work out nicely. You could even use fig to do. 

I would recommend using some for of proxy. Look at mine timhaak/proxy or https://github.com/jpetazzo/squid-in-a-can. It will reduce the chance of re-pulling files during the image rebuild. 

At a latter stage we can look at merging the plex and plexpass into one repo.

Also at some point may be nice to have different tag builds for different plex builds but I really don't have time to manage that and can't think of any way to full automate it simply.

**Comment by timhaak** [comment]
I'm going to close this for now. 

If there is something extra you want shout and we can re-open or create a new issue.

P.s also find it slow sometimes to work via github when talking you can email me directly at 'tim haak co uk'.  add relevant ats and dots where they would normally be.

**Comment by vvhof** [comment]
Do you even need a second git repository? 
Git repo:

timhaak/plex built from Dockerfile location /
timhaak/plexpass built from Dockerfile location (of the same repository) /plexpass/Dockerfile

**Comment by timhaak** [comment]
Hmm can't seem to find your comment on github. Assuming eventual consistency problem.




Not technically this could be run in branch. Though I like the idea that people who are having problems for the one can go to one repo as apposed to the other.




Also less likely to fix problems on the plexpass one as its in beta and more likely to be plex side.




Merging the two latter are quiet simple.




The problem with inheriting from the build is that you now would have the layer with the old not plex pass program still in it. Ie the image size is now larger for no benefit to someone wanting  plex pass.




I tend to only like inheriting when you have pieces that other images will be inheriting from. Ie one image adds node then you have images that use node inherit from it.




The updates that I'm talking about are from plex not the base images. 




If we only built when the underlying image rebuilt it would then only rebuild if they rebuild the debian image. 




What problem are you actually trying to solve? ie what pain are you feeling? 




Most of my images are built to to reduce admin on my side and be simple to use and know what's going on.  




I forgot to mention this https://github.com/timhaak/docker-plexpass-update it should do what your wanting with only updating the smallest possible amount. But you'll have to build it yourself as you need or add it to index under your name.






When I have more time. I'll create a base debian image that I'll update then build the others on top of that. Though that can also cause problems with two many layers (yes I have hit this).


—
Tim Haak

**Comment by vvhof** [comment]
Alright, fair points. Thanks Tim!
